### PR TITLE
Path whitespace changes.

### PR DESCRIPTION
**Pull request by nchikanov**
Change IsNullOrWhitespace to IsNullOrEmpty to be more consistent with legacy desktop .NET code. Path.GetFullPath no longer trims trailing whitespace. (Windows will still eat trailing spaces)

Test changes for CoreFX are done, will link in PR Request shortly. (https://github.com/dotnet/corefx/pull/22341)

### CODE REVIEW

**Comment by nchikanov** [comment]
See issue https://github.com/dotnet/corefx/issues/21358

File: src/mscorlib/src/System/IO/FileSystemEnumerable.cs [hunk 1]
```diff
@@ -496,9 +496,6 @@ private static String NormalizeSearchPattern(String searchPattern)
         {
             Contract.Requires(searchPattern != null);
 
-            // Win32 normalization trims only U+0020.
-            String tempSearchPattern = searchPattern.TrimEnd(PathInternal.s_trimEndChars);
-
             // Make this corner case more useful, like dir
             if (tempSearchPattern.Equals("."))
```
Comment (JeremyKuhne) [inline src/mscorlib/src/System/IO/FileSystemEnumerable.cs]: Should be searchPattern

**Comment by JeremyKuhne** [comment]
@dotnet-bot help

**Comment by JeremyKuhne** [comment]
@dotnet-bot test Windows_NT x64 corefx_baseline please
@dotnet-bot test Windows_NT x86 corefx_baseline please

**Pull request merged**